src/Microsoft.FeatureManagement/FeatureManager.cs

```diff
@@ -54,14 +65,106 @@ public Task<bool> IsEnabledAsync<TContext>(string feature, TContext appContext,
 
         public async IAsyncEnumerable<string> GetFeatureNamesAsync([EnumeratorCancellation] CancellationToken cancellationToken)
         {
-            await foreach (FeatureDefinition featureDefintion in _featureDefinitionProvider
-                                .GetAllFeatureDefinitionsAsync(cancellationToken)
-                                .ConfigureAwait(false))
+            await foreach (FeatureDefinition featureDefintion in _featureDefinitionProvider.GetAllFeatureDefinitionsAsync(cancellationToken).ConfigureAwait(false))
             {
                 yield return featureDefintion.Name;
             }
         }
 
+        public ValueTask<T> GetVariantAsync<T, TContext>(string feature, TContext appContext, CancellationToken cancellationToken)
+        {
+            return GetVariantAsync<T, TContext>(feature, appContext, true, cancellationToken);
+        }
+
+        public ValueTask<T> GetVariantAsync<T>(string feature, CancellationToken cancellationToken)
+        {
+            return GetVariantAsync<T, object>(feature, null, false, cancellationToken);
+        }
+
+        private async ValueTask<T> GetVariantAsync<T, TContext>(string feature, TContext appContext, bool useAppContext, CancellationToken cancellationToken)
+        {
+            if (feature == null)
+            {
+                throw new ArgumentNullException(nameof(feature));
+            }
+
+            FeatureVariant variant = null;
+
+            FeatureDefinition featureDefinition = await _featureDefinitionProvider.GetFeatureDefinitionAsync(feature, cancellationToken).ConfigureAwait(false);
+
+            if (featureDefinition != null)
+            {
+                if (string.IsNullOrEmpty(featureDefinition.Assigner))
+                {
+                    throw new FeatureManagementException(
+                        FeatureManagementError.InvalidConfiguration,
+                        $"Invalid feature variant assigner name for the feature {feature}");
```

Comment:
>Invalid [](http://example.com/codeflow?start=26&length=7)

A littler clearer about the error
Invalid... -> Missing...

Reply:
Updated.